pkg/devfile/adapters/kubernetes/utils/utils.go

```diff
@@ -32,14 +36,32 @@ func ConvertEnvs(vars []common.DockerimageEnv) []corev1.EnvVar {
 	return kVars
 }
 
+// ConvertPorts converts endpoint variables from the devfile structure to kubernetes ContainerPort
+func ConvertPorts(endpoints []common.DockerimageEndpoint) []corev1.ContainerPort {
+	containerPorts := []corev1.ContainerPort{}
+	for _, endpoint := range endpoints {
+		name := strings.TrimSpace(util.GetDNS1123Name(strings.ToLower(*endpoint.Name)))
+		containerPorts = append(containerPorts, corev1.ContainerPort{
+			Name:          name,
+			ContainerPort: *endpoint.Port,
+		})
+	}
+	return containerPorts
+}
+
 // GetContainers iterates through the components in the devfile and returns a slice of the corresponding containers
 func GetContainers(devfileObj devfile.DevfileObj) []corev1.Container {
 	var containers []corev1.Container
-	for _, comp := range adaptersCommon.GetSupportedComponents(devfileObj.Data) {
-		envVars := ConvertEnvs(comp.Env)
-		resourceReqs := GetResourceReqs(comp)
-		container := kclient.GenerateContainer(*comp.Alias, *comp.Image, false, comp.Command, comp.Args, envVars, resourceReqs)
-		containers = append(containers, *container)
+	// Only components with aliases are considered because without an alias commands cannot reference them
```

Comment:
Think you over wrote the master changes during rebase, this logic has been moved to common.utils.go GetSupportedComponents()

Reply:
Updated.